File: DesignPatternsInAutomatedTestingJava-Series/src/test/java/com/automatetheplanet/_1_page_object_pattern/BingTests.java
package com.automatetheplanet._1_page_object_pattern;

import com.automatetheplanet._1_page_object_pattern.Selenium.Bing.Pages.BingMainPage;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.annotations.AfterMethod;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

public class BingTests {
    public WebDriver driver;
    public WebDriverWait wait;

    @BeforeClass
    public static void classInit() {
        WebDriverManager.firefoxdriver().setup();
    }

    @BeforeMethod
    public void testInit() {
        driver = new FirefoxDriver();
        wait = new WebDriverWait(driver, 30);
    }

    @AfterMethod
    public void testCleanup() {
            driver.quit();
    }

    @Test
    public void searchTextInBing_First() {
        var bingMainPage = new BingMainPage(driver);
        bingMainPage.navigate();
        bingMainPage.search("Automate The Planet");
        bingMainPage.validateResultsCount("986,000 Results");
    }

    @Test
    public void searchTextInBing_Second() {
        var bingMainPage = new com.automatetheplanet._1_page_object_pattern.Pages.BingMainPage.BingMainPage(driver);
        bingMainPage.navigate();
        bingMainPage.search("Automate The Planet");
        bingMainPage.validate().resultsCount(",000 Results");
    }
}

File: dotnet/Design-Architecture-Series/PageObjectPattern/BingTests.cs
// <copyright file="BingTests.cs" company="Automate The Planet Ltd.">
// Copyright 2021 Automate The Planet Ltd.
// Licensed under the Apache License, Version 2.0 (the "License");
// You may not use this file except in compliance with the License.
// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// </copyright>
// <author>Anton Angelov</author>
// <site>http://automatetheplanet.com/</site>
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Support.UI;
using PageObjectPattern.Selenium.Bing.Pages;

namespace PageObjectPattern
{
    [TestClass]
    public class BingTests
    {
        public IWebDriver Driver { get; set; }
        public WebDriverWait Wait { get; set; }

        [TestInitialize]
        public void SetupTest()
        {
            Driver = new FirefoxDriver();
            Wait = new WebDriverWait(Driver, TimeSpan.FromSeconds(30));
        }

        [TestCleanup]
        public void TeardownTest()
        {
            Driver.Quit();
        }

        [TestMethod]
        public void SearchTextInBing_First()
        {
            var bingMainPage = new BingMainPage(Driver);
            bingMainPage.Navigate();
            bingMainPage.Search("Automate The Planet");
            bingMainPage.ValidateResultsCount(",000 Results");
        }

        [TestMethod]
        public void SearchTextInBing_Second()
        {
            var bingMainPage = new Pages.BingMainPage(Driver);
            bingMainPage.Navigate();
            bingMainPage.Search("Automate The Planet");
            bingMainPage.Validate().ResultsCount(",000 Results");
        }
    }
}